UIKitMovie/UIKitMovie/Presenter/MoviePresenter.swift

```diff
@@ -0,0 +1,68 @@
+// MoviePresenter.swift
+// Copyright © RoadMap. All rights reserved.
+
+import Foundation
+
+/// Представление списка фильмов
+final class MoviePresenter: MovieViewPresenterProtocol {
+    // MARK: - Private Constants
+
+    private enum Constants {
+        static let topLevelJSON = "results"
+    }
+
+    // MARK: - Public Properties
+
+    var movies: [Movie]?
+
+    // MARK: - Private Properties
+
+    private let networkService: NetworkServiceProtocol!
+    private var router: RouterProtocol?
+    private weak var view: MovieViewProtocol?
+
+    // MARK: - Initializers
+
+    required init(view: MovieViewProtocol, networkService: NetworkServiceProtocol, router: RouterProtocol) {
+        self.view = view
+        self.networkService = networkService
+        self.router = router
+        getMovie(index: 0)
+    }
+
+    // MARK: - Public Methods
+
+    func tapOnFilm(film: Movie?) {
+        router?.showFilm(film: film)
+    }
+
+    func getMovie(index: Int) {
+        switch index {
+        case 0:
+            fetchMovies(method: MethodType.upcomingMethod)
+        case 1:
+            fetchMovies(method: MethodType.popularMethod)
+        case 2:
+            fetchMovies(method: MethodType.topRatedMethod)
+        default:
+            break
+        }
+    }
+
+    // MARK: - Private Methods
+
+    private func fetchMovies(method: MethodType) {
+        networkService.fetchMovies(method: method) { [weak self] result in
+            guard let self = self else { return }
+            DispatchQueue.main.async {
```

Comment:
```suggestion
            DispatchQueue.main.async {
```
 проверь мы точно  не на главном потоке